source/fe/fe_tools_interpolate.cc

```diff
@@ -222,7 +222,8 @@ namespace FETools
     for (types::global_dof_index i=0; i<dof2.n_dofs(); ++i)
       if (locally_owned_dofs.is_element(i))
         {
-          Assert(touch_count(i) != typename OutVector::value_type(),
+          const typename OutVector::value_type touch_i = touch_count(i);
+          Assert(touch_i != typename OutVector::value_type(0),
```

Comment:
You'll get a warning about an unused variable here in release mode. Please just do the cast by hand inside the assertion.